src/Illuminate/Filesystem/FilesystemAdapter.php

```diff
@@ -143,7 +143,12 @@ public function putFile($path, $file, $options = [])
      */
     public function putFileAs($path, $file, $name, $options = [])
     {
-        $stream = fopen($file->getRealPath(), 'r+');
+        // If you server is under Windows and getRealPath() is false we use getPathName() instead.
+        if($file->getRealPath()) {
+            $stream = fopen($file->getRealPath(), 'r+');
+        } else {
+            $stream = fopen($file->getPathName(), 'r+');
+        }
```

Comment:
It might be better to explicitly check if you're on Windows before applying this behavior so it doesn't cascades to other systems. Perhaps go as far as to check to which version this behavior needs to be applied to.